src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -243,10 +243,18 @@ public static long ToOACurrency(decimal value)
         // The possible binary representations of a particular value are all
         // equally valid, and all are numerically equivalent.
         //
-        public Decimal(int[] bits)
+        public Decimal(int[] bits) :
+            this(bits != null ? bits.AsSpan() : throw new ArgumentNullException(nameof(bits)))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="decimal"/> to a decimal value represented in binary and contained in the specified span.
+        /// </summary>
+        /// <param name="bits">A span of four <see cref="int"/>s containing a binary representation of a decimal value.</param>
+        /// <exception cref="ArgumentException">The length of <paramref name="bits"/> is not 4, or the representation of the decimal value in <paramref name="bits"/> is not valid.</exception>
+        public Decimal(ReadOnlySpan<int> bits)
         {
-            if (bits == null)
-                throw new ArgumentNullException(nameof(bits));
             if (bits.Length == 4)
```

Comment:
Do we know why this is doing `if (bits.Length == 4) { code } throw` rather than `if (bits.Length != 4) { throw} code;`?
Most of the framework does the latter, it reduces the overall nesting, and removes the need for the early return statement.

Reply:
Presumably because of that IsValid check below?

Reply:
ah right, my brain skimmed over that bit.